utils/rigid_origami_utils.py

```diff
@@ -0,0 +1,506 @@
+                    e_cr_idx = cls.crease_lines.edges.index(obj_e)
+                    if e_cr_idx not in cls.inside_vertices[iv_idx].crease_indices:
+                        continue
+                    e_l_idx = cls.inside_vertices[iv_idx].crease_indices.index(e_cr_idx)
+                    sign = 1
+                    if abs(e_l_idx - local_idx) > 1:
+                        sign = -1 if local_idx == 0 else 1
+                    else:
+                        sign = e_l_idx - local_idx
+
+                    rad = rad * sign
+                    break
+        else:
+            # Center of the target polygon
+            center = face.calc_center_median()
+            poly_center = np.array([center[0], center[1], center[2]])
+            # Cross product between 'edge[1] - edge[0]' and 'center - edge[0]'
+            va = poly_center - verts[edge[0]]
+            vb = verts[edge[1]] - verts[edge[0]]
+            n = np.cross(va, vb)
+            n = n / np.linalg.norm(n)
+            face_n = np.array([face.normal[0], face.normal[1], face.normal[2]])
+
+            rad = 0.0
+            if cls.crease_lines.edges.count(edge) > 0:
+                rad = cls.crease_lines.delta_angles[cls.crease_lines.edges.index(edge)]
+                rad *= -1 if np.dot(n, face_n) > 0 else 1
+            return vb, rad, verts[edge[0]]
+            
+                        
+        return vec, rad, v0
+
+    # Function to rotate a face
+    @classmethod
+    def __rotate_face(cls, face_rot):
+        v_indices = [v.index for v in face_rot.face.verts]
+        
+        rotated_verts = []
+        for v in face_rot.face.verts:
+            # Rotate source vertex using quaternion            
+            source_q = np.array([cls.obj.verts[v.index][0], cls.obj.verts[v.index][1], cls.obj.verts[v.index][2], 1])
+            vq = np.dot(face_rot.rot_quat, source_q.T)
+            v_result = np.array([vq[0], vq[1], vq[2]])
+            rotated_verts.append(v_result)
+            
+        return v_indices, rotated_verts
+    
+    @classmethod
+    # Function to create rotation quaternion
+    # to rotate 'rad' radian around 'n' axis
+    def __rot_quat(cls, rad, n):
+        n = n / np.linalg.norm(n)
+        rad = float(rad)
+        R = np.array([[np.cos(rad)+n[0]*n[0]*(1-np.cos(rad)), \
+                       n[0]*n[1]*(1-np.cos(rad))-n[2]*np.sin(rad), \
+                       n[0]*n[2]*(1-np.cos(rad))+n[1]*np.sin(rad), 0], \
+                      [n[1]*n[0]*(1-np.cos(rad))+n[2]*np.sin(rad), \
+                       np.cos(rad)+n[1]*n[1]*(1-np.cos(rad)), \
+                       n[1]*n[2]*(1-np.cos(rad))-n[0]*np.sin(rad), 0], \
+                      [n[2]*n[0]*(1-np.cos(rad))-n[1]*np.sin(rad), \
+                       n[2]*n[1]*(1-np.cos(rad))+n[0]*np.sin(rad), \
+                       np.cos(rad)+n[2]*n[2]*(1-np.cos(rad)), 0], \
+                       [0, 0, 0, 1]])
+        return R
```

Comment:
```suggestion
    def __rot_quat(cls, rad, n):
        n = n / np.linalg.norm(n)
        rad = float(rad)
        cos_rad = np.cos(rad)
        sin_rad = np.sin(rad)
        R = np.array([
                       [cos_rad + n[0] * n[0] * (1 - cos_rad), \
                       n[0] * n[1] * (1 - cos_rad) - n[2] * sin_rad , \
                       n[0] * n[2] * (1 - cos_rad) + n[1] * sin_rad , 0], \
                       [n[1] * n[0] * (1 - cos_rad)+ n[2] * sin_rad , \
                       cos_rad + n[1] * n[1] * (1-cos_rad ), \
                       n[1] * n[2] * (1 - cos_rad) - n[0] * sin_rad , 0], \
                       [n[2] * n[0] * (1 - cos_rad) - n[1] * sin_rad , \
                       n[2] * n[1] * (1 - cos_rad) + n[0] * sin_rad , \
                       cos_rad + n[2] * n[2] * (1 - cos_rad), 0], \
                       [0, 0, 0, 1]])
        return R
```

Reply:
Thank you. It had not been easy to read, but your modification made it really simple and readable!